chain/consensus/mir/mempool/fifo/fifo_pool.go

```diff
@@ -0,0 +1,49 @@
+package fifo
+
+import (
+	mirrequest "github.com/filecoin-project/mir/pkg/pb/requestpb"
+)
+
+// Pool is a structure to implement the simplest pool that enforces FIFO policy on client messages.
+// When a client sends a message we add clientID to orderingClients map and clientByCID.
+// When we receive a message we find the clientID and remove it from orderingClients.
+// We don't need using sync primitives since the pool's methods are called only by one goroutine.
+type Pool struct {
+	clientByCID     map[string]string // messageCID -> clientID
```

Comment:
@dnkolegov, why do we use `string` instead of `cid` for the key in this map?

Reply:
> Also, the client IDs could probably be mirtype.ClientID...
Because ClientID in Mir's Request is a string. So there is not much sense to use NodeID here. This only adds the Mir as a dependency into the pool package

Reply:
Ok, fair enough. I still think it's cleaner to convert it go `ClientID`, because that's its true type (while `string` just happens to be its protobuf representation), but I don't have too strong feelings about this.

Reply:
> why do we use string instead of cid for the key in this map?

Fixed